piplapis/data/fields.py

```diff
@@ -468,6 +468,90 @@ def display(self):
         return self.address or self.address_md5
 
 
+class Vehicle(Field):
+    """
+        Vehicle information.
+    """
+
+    children = ('vin', 'year', 'make', 'model', 'color', 'vehicle_type')
+
+    def __init__(
+            self,
+            vin=None,
+            year=None,
+            make=None,
+            model=None,
+            color=None,
+            vehicle_type=None,
+            *args, **kwargs):
+
+        Field.__init__(self, *args, **kwargs)
+        self.vin = vin
+        self.year = year
+        self.make = make
+        self.model = model
+        self.color = color
+        self.vehicle_type = vehicle_type
+
+    @property
+    def display(self) -> str:
+        make = f"{self.make.title()} " if self.make else ""
+        model = f"{self.model.title()} " if self.model else ""
+        make_and_model = f"{make}{model}" if make or model else ""
+        year = f"{self.year} " if self.year else ""
+        vin = self.vin or ""
+        color = f", {self.color.title()}" if self.color else ""
+        vehicle_type = f"({self.vehicle_type.title()})" if self.vehicle_type else ""
+        type_and_color = f"{vehicle_type}{color} " if vehicle_type or color else ""
+        hyphen = "- " if year or make_and_model or vehicle_type or color else ""
+
+        return f"{year}{make_and_model}{type_and_color}{hyphen}VIN {vin}"
+
+    def validate_vin(self, vin: str) -> bool:
```

Comment:
[according to this](https://stackoverflow.com/a/50262940/14004541),
seems like it requires some changes.
i'm usually very pro-type-hints, but since the existing code doesn't have any, it might not be worth it in this case...

Reply:
Definitely make sense.
Let's remember for the next release, if we decide to go with type hint, let's do it across the project.